packages/wallet/src/wallet.ts

```
@@ -1,40 +1,172 @@
+                return {
+                    isValid: false,
+                    error: `Invalid mnemonic: expected 12 or 24 words, got ${wordCount}`
+                };
+            }
+
+            // Validate proper BIP-39 words (in a real implementation)
+            // For this example, we just check if any word is too short or contains non-alphabet chars
+            for (const word of words) {
+                if (word.length < 3) {
+                    return {
+                        isValid: false,
+                        error: `Invalid word in mnemonic: "${word}" is too short`
+                    };
+                }
+
+                if (!/^[a-zA-Z]+$/.test(word)) {
+                    return {
+                        isValid: false,
+                        error: `Invalid word in mnemonic: "${word}" contains non-alphabet characters`
+                    };
+                }
+            }
 
-        return new Wallet(core, wallet, password)
+            return { isValid: true };
+        } catch (error) {
+            return {
+                isValid: false,
+                error: error instanceof Error ? error.message : 'Unknown error validating mnemonic'
+            };
+        }
     }
 
+    /**
+     * Restore a wallet from an existing mnemonic
+     * @param core WalletCore instance
+     * @param mnemonic Mnemonic phrase (12 or 24 words)
+     * @param password Password for encryption
+     * @returns A restored Wallet instance
+     */
     static restore(core: WalletCore, mnemonic: string, password: string): Wallet {
-        let wallet = core.HDWallet.createWithMnemonic(mnemonic, "");
+        // Validate mnemonic
+        const validation = Wallet.validateMnemonic(mnemonic);
+        if (!validation.isValid) {
+            throw new Error(validation.error);
+        }
 
-        return new Wallet(core, wallet, password)
+        try {
+            let wallet = core.HDWallet.createWithMnemonic(mnemonic, '');
```

I believe the mnemonic validator checks this internally. We can add some "learning tests" here  to confirm it.  
Try to restore a wallet with an invalid mnemonic. TrustWallet Core should return an error

---

i trying by test the invalid mnemonic. The Wallet Core throw the error 
``` 
    Error: Failed to restore wallet: Unknown error
```